parser/src/dataset/write.rs

```diff
@@ -106,41 +179,45 @@ where
             to,
             encoder,
             SpecificCharacterSet::default(),
+            options,
         ))
     }
 
     /// Create a new data set writer
-    /// with the given transfer syntax specifier
-    /// and the specific character set to assume by default.
-    ///
-    /// Note that the data set being written
-    /// can override the character set with the presence of a
-    /// _Specific Character Set_ data element.
-    pub fn with_ts_cs(to: W, ts: &TransferSyntax, charset: SpecificCharacterSet) -> Result<Self> {
+    /// with the given transfer syntax specifier,
+    /// specific character set and options.
+    pub fn with_ts_cs_options(
+        to: W,
+        ts: &TransferSyntax,
+        charset: SpecificCharacterSet,
+        options: DataSetWriterOptions,
+    ) -> Result<Self> {
         let encoder = ts.encoder_for().context(UnsupportedTransferSyntaxSnafu {
             ts_uid: ts.uid(),
             ts_alias: ts.name(),
         })?;
-        Ok(DataSetWriter::new_with_codec(to, encoder, charset))
+        Ok(DataSetWriter::new_with_codec(to, encoder, charset, options))
     }
 }
 
 impl<W, E> DataSetWriter<W, E> {
-    pub fn new(to: W, encoder: E) -> Self {
+    pub fn new(to: W, encoder: E, options: DataSetWriterOptions) -> Self {
```

Comment:
This is a breaking change, but it can be worked around by adding `new_with_options` instead of changing `new`.